server/test/src/main/java/org/eclipse/lsp/cobol/test/engine/UseCasePreprocessorListener.java

```diff
@@ -213,27 +213,32 @@ public void enterParagraphStatement(ParagraphStatementContext ctx) {
   @Override
   public void exitParagraphStatement(ParagraphStatementContext ctx) {
     pop();
-    ofNullable(ctx.paragraphUsage())
-            .map(p -> p.paragraph)
-            .ifPresent(
-                    it ->
-                            processToken(
-                                    it.identifier().getText(),
-                                    ctx,
-                                    it.replacement(),
-                                    paragraphUsages,
-                                    ctx.diagnostic()));
-    ofNullable(ctx.paragraphDefinition())
-            .map(ParagraphDefinitionContext::word)
-            .ifPresent(
-                    it ->
-                            processToken(
-                                    it.identifier().getText(),
-                                    ctx,
-                                    it.replacement(),
-                                    paragraphDefinitions,
-                                    ctx.diagnostic()));
-  }
+    ParagraphUsageContext p = ctx.paragraphUsage();
+    if (p != null && p.paragraph != null) {
+      String section = p.section == null
+              ? null
+              : getReplacementText(p.section.identifier().getText(), p.section.replacement()).get(0);
+
+      processProcedureToken(
+              p.paragraph.identifier().getText(),
+              ctx, p.paragraph.replacement(),
+              procedureUsages,
+              ctx.diagnostic(),
+              () -> new ProcedureId(section,
+                      getReplacementText(p.paragraph.getText(), p.paragraph.replacement()).get(0).toUpperCase()));
```

Comment:
The lambda is unconditionally called in the `processProcedureToken`, why not simply pass the new ProcedureId object?